### PR TITLE
update shaman on build status

### PR DESCRIPTION
**Pull request by alfredodeza**
Includes a  couple of helper scripts to avoid repetitive shell-to-json-and-back

### CODE REVIEW

File: scripts/build_utils.sh [hunk 1]
```diff
@@ -128,3 +128,78 @@ check_binary_existence () {
...
+    distro_version=$5
+    distro_arch=$6
+    cat > $WORKSPACE/build_status.json << EOF
+{
+    "extra":{
+        "version":"$vers",
+        "root_build_cause":"$ROOT_BUILD_CAUSE",
+        "node_name":"$NODE_NAME",
+        "job_name":"$JOB_NAME",
+        "url":"$BUILD_URL",
```
Comment (andrewschoen) [inline scripts/build_utils.sh]: ``url`` should not be in ``extra``

File: scripts/build_utils.sh [hunk 2]
```diff
@@ -128,3 +128,78 @@ check_binary_existence () {
...
+        "build_user":"$BUILD_USER",
+        "log_url":"$BUILD_URL/consoleFull"
```
Comment (andrewschoen) [inline scripts/build_utils.sh]: ``log_url`` should not be in ``extra``

File: scripts/build_utils.sh [hunk 3]
```diff
@@ -128,3 +128,78 @@ check_binary_existence () {
...
+    },
+    "state":"$state",
+    "distro":"$distro",
+    "distro_version":"$distro_version",
+    "distro_arch":"$distro_arch",
```
Comment (andrewschoen) [inline scripts/build_utils.sh]: I didn't add a ``distro_arch`` field to my shaman PR, I'll need to fix that

File: scripts/build_utils.sh [hunk 4]
```diff
@@ -128,3 +128,78 @@ check_binary_existence () {
...
+    "branch":"$BRANCH",
```
Comment (andrewschoen) [inline scripts/build_utils.sh]: Is this the parsed version of branch that removes any extra forward slashes in the branch name? I think we'd want to get this the same as what gets used in chacra.

File: ceph-dev-build/build/build_deb [hunk 5]
```diff
@@ -67,6 +67,14 @@ gen_debian_version() {
 
 bpvers=`gen_debian_version $debian_version $DIST`
 
+# Normalize variables across rpm/deb builds
+NORMAL_DISTRO=$distro
+NORMAL_DISTRO_VERSION=$DIST
+NORMAL_ARCH=$ARCH
+
+# update shaman with the build status
+update_build_status "started" "ceph" $NORMAL_DISTRO $NORMAL_DISTRO_VERSION $NORMAL_ARCH
```
Comment (andrewschoen) [inline ceph-dev-build/build/build_deb]: These initial calls need to use ``POST`` not ``PUT``.

File: scripts/build_utils.sh (hunk 4 above)
Comment (alfredodeza) [inline scripts/build_utils.sh]: correct, it is the filtered one

File: scripts/build_utils.sh [hunk 6]
```diff
@@ -128,3 +128,97 @@ check_binary_existence () {
...
+    project=$1
+
+    state="failed"
+
+    # optional
+    distro=$2
+    distro_version=$3
+    distro_arch=$4
+
+    submit_build_status "POST" $state $project $distro $distro_version $distro_arch
```
Comment (andrewschoen) [inline scripts/build_utils.sh]: Looks like you are missing your closing ``}`` here

File: scripts/build_utils.sh [hunk 7]
```diff
@@ -128,3 +128,97 @@ check_binary_existence () {
...
+        "build_user":"$BUILD_USER"
+    },
+    "url":"$BUILD_URL",
+    "log_url":"$BUILD_URL/consoleFull",
+    "state":"$state",
+    "distro":"$distro",
+    "distro_version":"$distro_version",
+    "distro_arch":"$distro_arch",
+    "branch":"$BRANCH",
+    "sha1":"$GIT_COMMIT",
```
Comment (andrewschoen) [inline scripts/build_utils.sh]: ``$GIT_COMMIT`` is only available to the ``ceph-dev-setup`` job, it's passed to ``ceph-dev-build`` as ``$SHA1``.

**Pull request merged**